### PR TITLE
Speedy pass

### PR DESCRIPTION
**Pull request by derp00monsta**
speedy-pass is now speedy passing and it doesn't seem to be messing with shooting into the speaker, but that's only in sim.

### CODE REVIEW

**Review by Hadizah454 — COMMENTED** [review]
It looks good to me and the images are a nice touch. There seems to be a merge conflict in the piece control class but I'll resolve it myself so you don't need to worry about that.

**Review by Hadizah454 — APPROVED** [review]
Actually nvm @derp00monsta can you try to resolve the merge conflict yourself because idk why but my computer is freaking out on me right now sorry.

**Comment by GalexY727** [comment]
I feel that there is a better way to do this rather than adding alignmentcmds into piececontrol, because that turns piececontrol into a second-hand "robotcontainer" type of file. I think that there should be some sort of method that we can get from alignmentcmds and pass into piececontrol rather than sending the entire file into inheritance. 

That was a lot of words and if I didn't type that out quite right let me know and I'll try to reword it.

**Comment by derp00monsta** [comment]
> I feel that there is a better way to do this rather than adding alignmentcmds into piececontrol, because that turns piececontrol into a second-hand "robotcontainer" type of file. I think that there should be some sort of method that we can get from alignmentcmds and pass into piececontrol rather than sending the entire file into inheritance.
> 
> That was a lot of words and if I didn't type that out quite right let me know and I'll try to reword it.

How do we use a method from alignmentcmds w/out importing the file into piececontrol? None of the classes imported into piececontrol on main seem to have alignmentcmds in them either.
I think I might understand I'm just not sure how to do what you are suggesting.

**Comment by GalexY727** [comment]
> > I feel that there is a better way to do this rather than adding alignmentcmds into piececontrol, because that turns piececontrol into a second-hand "robotcontainer" type of file. I think that there should be some sort of method that we can get from alignmentcmds and pass into piececontrol rather than sending the entire file into inheritance.
> > That was a lot of words and if I didn't type that out quite right let me know and I'll try to reword it.
> 
> How do we use a method from alignmentcmds w/out importing the file into piececontrol? None of the classes imported into piececontrol on main seem to have alignmentcmds in them either. I think I might understand I'm just not sure how to do what you are suggesting.

We are going from full inheritance to just partial. If all we need is one method from alignmentcmds, we can give that method to piececontrol whenever we call the command, so robotcontainer is in fuller control over the scope of piececontrol.

**Comment by GalexY727** [comment]
one thing i would like to mention with this pr is that it removed the feature of waiting for hdc to be rotated before shooting, and although it was reintroduced inside of [9f96b1e](https://github.com/Patribots4738/Crescendo2024/pull/282/commits/9f96b1e990f8c773f060ca3dd8969a060b644210) i worry that there may be other features that got overlooked somehow.

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 1]
```diff
@@ -24,6 +24,9 @@ public class Pivot extends SubsystemBase implements Logged {
 	@Log
 	private boolean atDesiredAngle = false;
 
+    @Log
```
Comment (GalexY727) [inline src/main/java/frc/robot/subsystems/Pivot.java:27]: this formatting is bugged in github but looks fine in the code

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 2]
```diff
@@ -110,4 +116,8 @@ public double getTargetAngle() {
 	public boolean getAtDesiredAngle() {
 		return atDesiredAngle;
 	}
+
+    public boolean getAtDesiredPassAngle() {
+        return atDesiredPassAngle;
+    }
```
Comment (GalexY727) [inline src/main/java/frc/robot/subsystems/Pivot.java:122]: this formatting is bugged in github but looks fine in the code

File: src/main/java/frc/robot/commands/managers/PieceControl.java [hunk 3]
```diff
@@ -99,8 +99,18 @@ public Command brakeIntakeAndIndexer() {
 
     // TODO: only run angle reset when we are not using prepareSWDCommand
     public Command shootWhenReady(Supplier<Pose2d> poseSupplier, Supplier<ChassisSpeeds> speedSupplier, BooleanSupplier atDesiredAngle) {
-        return Commands.waitUntil(() -> shooterCmds.shooterCalc.readyToShootSupplier().getAsBoolean() && atDesiredAngle.getAsBoolean())
-                .andThen(noteToShoot(poseSupplier, speedSupplier));
+        return
+            new ActiveConditionalCommand(
```
Comment (PatribotsProgramming) [inline src/main/java/frc/robot/commands/managers/PieceControl.java:103]: This doesn't need to be active, it should just be a either

**Review by PatribotsProgramming — CHANGES_REQUESTED** [review]
A small formatting fix and it would be nice to be able to test the limits of the deadband to see what the biggest amount of change we are allowing before we send it an a full comp

**Review by PatribotsProgramming — APPROVED** [review]
If there are complications with our passing at comp then we can revert this pr woohoo1!!

**Pull request merged**